src/main/java/com/minecolonies/colony/buildings/BuildingBuilder.java

```diff
@@ -44,6 +48,28 @@ public AbstractJob createJob(CitizenData citizen)
         return new JobBuilder(citizen);
     }
 
+    /**
+     * 
+     * @see AbstractBuilding#onUpgradeComplete(int)
+     */
+    @Override
+    public void onUpgradeComplete(final int newLevel)
+    {
+        super.onUpgradeComplete(newLevel);
+
+        final EntityPlayer owner = ServerUtils.getPlayerOnServerFromUUID(getColony().getPermissions().getOwner());
```

Comment:
Or make another method `permission.getOwnerAsPlayer()´? Currently it returns the UUID